docs/articles/documentation/test-api/handling-native-dialogs.md

```diff
@@ -0,0 +1,141 @@
+
+Dialog Type  | Return Value                                             | Default Handling
+------------ | -------------------------------------------------------- | --------------
+alert        | Ignored                                                  | 'OK' button.
+beforeunload | Ignored                                                  | 'Leave' button. There is no way to emulate 'Stay' programmatically.
+confirm      | `true` to answer 'Yes'; `false` to answer 'No'.          | 'No' button.
+prompt       | A string that contains text to be typed into the prompt. | 'Cancel' button.
+
+The following example shows how to handle an alert dialog.
+
+```js
+test('My test', async t => {
+    await t
+        .setNativeDialogHandler(() => true)
+        .click('#show-alert-button');
+});
+```
+
+The next example demonstrates a test that handles two confirm dialogs and an alert dialog.
+
+```js
+test('My test', async t => {
+    await t
+        .setNativeDialogHandler((type, text) => {
+            if (type === 'confirm') {
+                switch (text) {
+                    case 'Do you want to subscribe?':
+                        return false;
+                    case 'Do you want to delete your account?':
+                        return true;
+                }
+            }
+            return true;
+        })
+        .click('#confirm-subscription');
+        .click('#show-alert');
+        .click('#confirm-account-deletion');
+});
+```
+
+## Dialog History
+
+You can get the history of invoked native dialogs to check if a certain dialog appeared or not.
+
+Use the [test controller's](test-code-structure.md#test-controller) `getNativeDialogHistory` method for this.
+
+```text
+t.getNativeDialogHistory() → [{type, text}]
+```
+
+`t.getNativeDialogHistory` returns a stack of history entries (i.e. an array in which the latest dialog has an index of `0`).
+Each entry corresponds to a certain native dialog and contains the following properties.
+
+Property | Type   | Description
+-------- | ------ | -------------
+`type`   | String | The type of the native dialog. `'alert'` &#124; `'confirm'` &#124; `'beforeunload'` &#124; `'prompt'`.
+`text`   | String | Text of the dialog message.
+
+```js
+import { expect } from 'chai';
+
+fixture `My fixture`
+    .page('http://www.example.com/');
+
+test('My test', async t => {
+    await t
+        .setNativeDialogHandler((type, text) => {
+            switch (type) {
+                case 'confirm':
+                    return true;
+                case 'prompt':
+                    return 'Hello!';
+            }
+        })
+        .click('#show-alert')
+        .click('#show-confirm')
+        .click('#show-prompt');
+
+    const history = await t.getNativeDialogHistory();
+
+    expect(history[0].type).to.equal('prompt');
+    expect(history[0].text).to.equal('say hello');
+    expect(history[1].type).to.equal('confirm');
+    expect(history[2].type).to.equal('alert');
+});
```

Comment:
Yes. Most common scenario is to check the latest entry in the history, so you will write:
```js
history[0]
```
instead of
```js
history[history.length - 1]
```